packages/design/src/Form/components/SocialSecurityNumber/SocialSecurityNumber.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import classNames from 'classnames';
+import { useFormContext } from 'react-hook-form';
+import { type SocialSecurityNumberProps } from '@atj/forms';
+
+import { type PatternComponent } from '../../index.js';
+
+export const SocialSecurityNumberPattern: PatternComponent<
+  SocialSecurityNumberProps
+> = ({ ssnId, hint, label, required, error, value }) => {
+  const { register } = useFormContext();
+  const errorId = `input-error-message-${ssnId}`;
+
+  return (
+    <fieldset className="usa-fieldset">
+      <div className={classNames('usa-form-group margin-top-2')}>
+        <label
+          className={classNames('usa-label', {
+            'usa-label--error': error,
+          })}
+          htmlFor={ssnId}
+        >
+          {label || 'Social Security Number'}
+          {required && <span className="required-indicator">*</span>}
+        </label>
+        {hint && (
+          <div className="usa-hint" id={`hint-${ssnId}`}>
+            {hint}
+          </div>
+        )}
+        {error && (
+          <div className="usa-error-message" id={errorId} role="alert">
+            {error.message}
+          </div>
+        )}
+        <input
+          className={classNames('usa-input usa-input--xl', {
+            'usa-input--error': error,
+          })}
+          id={ssnId}
+          type="text"
+          defaultValue={value}
+          {...register(ssnId, { required })}
+          aria-describedby={error ? `${ssnId} ${errorId}` : ssnId}
```

Comment:
Thank you! I am just curious to know why do you think it should be hint? 🤔 when I used `ssnId` I referred to the input Id on line 40. Also hint doesn't get an error. Error message is separate from the hint message. Please let me know if I missing something here.

Reply:
Great question! When you add `aria-describedby`, that is saying that there is an additional element somewhere else on the page that provides information about the element where the `aria-describedby` appears. The element with error on lines 31-35 describes why the input is invalid, and the element with the hint on line 26-30 describes what type of information we're looking for in the input. 

It wouldn't be `ssnId` on line 40 because that's the same element that has the `aria-describedby` attribute and like saying that the input describes itself.

Reply:
Thank you for the clarification, got it now! Updated the code accordingly.